### PR TITLE
[extension/googleclientauth] Update github.com/GoogleCloudPlatform/opentelemetry-operations-go v0.48.0

### PR DESCRIPTION
**Pull request by rinx**
**Description:** <Describe what has changed.>

Update github.com/GoogleCloudPlatform/opentelemetry-operations-go/extension/googleclientauthextension to v0.48.0.

This update includes support for Google-signed ID token.
https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/releases/tag/v0.48.0

**Link to tracking Issue:** <Issue number if applicable>

#33185

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

The documents about googleclientauth extension also updated.

### CODE REVIEW

**Comment by dashpole** [comment]
cc @damemi @psx95 
@psx95 Can you coordinate with @rinx for the update?

xref: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33483

**Review by psx95 — APPROVED** [review]

**Review by songy23 — COMMENTED** [review]
Needs `make gotidy`, probably `make genotelcontribcol` too

I guess this needs a changelog too?

**Comment by rinx** [comment]
@psx95 Thanks. I'll try to update the tests.

**Comment by rinx** [comment]
@psx95 Could you please review the changes?

File: extension/googleclientauthextension/metadata.yaml
Comment (psx95) [inline extension/googleclientauthextension/metadata.yaml:12]: I'm not familiar with the use of `metadata.yaml` file in this repo, so just confirming - is this required when the `generated_package_test.go` already programmatically seem to ignore the goleak coming from `opencensus` ?

**Review by psx95 — COMMENTED** [review]
LGTM,  pending one clarification.

PS - Just calling out that the kubernetes test seem to be failing, but I don't see how this PR could cause them to fail.

File: extension/googleclientauthextension/metadata.yaml
Comment (rinx) [reply extension/googleclientauthextension/metadata.yaml:12]: `generated_package_test.go` is generated based on `metadata.yaml`. It is required to write `goleak.ignore` block here.

Without this `goleak.ignore`, the tests should fail with goroutine leak caused by `opencensus`.

**Comment by rinx** [comment]
> PS - Just calling out that the kubernetes test seem to be failing, but I don't see how this PR could cause them to fail.

I think this is related to #33520 and #33521.

File: extension/googleclientauthextension/metadata.yaml
Comment (psx95) [reply extension/googleclientauthextension/metadata.yaml:12]: I see, thanks for the confirmation.

**Review by psx95 — APPROVED** [review]

**Review by dashpole — APPROVED** [review]

**Review by codeboten — COMMENTED** [review]
Sorry @rinx this will need another rebase after the v0.103.0 release. Please wait until the release is out, and rebase the PR to address the conflicts and we can get this merged

**Review by codeboten — COMMENTED** [review]
@rinx this is ready to rebase whenever you have a moment

**Comment by rinx** [comment]
@codeboten I rebased it. Thanks.

**Review by codeboten — APPROVED** [review]
Thanks @rinx

**Pull request merged**
